src/common/webrequest_curl.cpp

```diff
@@ -0,0 +1,559 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        src/common/webrequest_curl.h
+// Purpose:     wxWebRequest implementation using libcurl
+// Author:      Tobias Taschner
+// Created:     2018-10-25
+// Copyright:   (c) 2018 wxWidgets development team
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#if defined(__BORLANDC__)
+#pragma hdrstop
+#endif
+
+#include "wx/webrequest.h"
+
+#if wxUSE_WEBREQUEST_CURL
+
+#include "wx/webrequest_curl.h"
+
+#ifndef WX_PRECOMP
+    #include "wx/log.h"
+    #include "wx/utils.h"
+#endif
+
+// Define symbols that might be missing from older libcurl headers
+#ifndef CURL_AT_LEAST_VERSION
+#define CURL_VERSION_BITS(x,y,z) ((x)<<16|(y)<<8|z)
+#define CURL_AT_LEAST_VERSION(x,y,z) \
+  (LIBCURL_VERSION_NUM >= CURL_VERSION_BITS(x, y, z))
+#endif
+
+#if CURL_AT_LEAST_VERSION(7, 28, 0)
+    #define wxCURL_HAVE_MULTI_WAIT 1
+#else
+    #define wxCURL_HAVE_MULTI_WAIT 0
+#endif
+
+//
+// wxWebResponseCURL
+//
+
+static size_t wxCURLWriteData(void* buffer, size_t size, size_t nmemb, void* userp)
+{
+    if ( userp )
+        return static_cast<wxWebResponseCURL*>(userp)->WriteData(buffer, size * nmemb);
+    else
+        return 0;
+}
+
+static size_t wxCURLHeader(char *buffer, size_t size, size_t nitems, void *userdata)
+{
+    if ( userdata )
+        return static_cast<wxWebResponseCURL*>(userdata)->AddHeaderData(buffer, size * nitems);
+    else
+        return 0;
+}
+
+wxWebResponseCURL::wxWebResponseCURL(wxWebRequest& request) :
+    wxWebResponse(request)
+{
+    curl_easy_setopt(GetHandle(), CURLOPT_WRITEDATA, static_cast<void*>(this));
+    curl_easy_setopt(GetHandle(), CURLOPT_HEADERDATA, static_cast<void*>(this));
+}
+
+size_t wxWebResponseCURL::WriteData(void* buffer, size_t size)
+{
+    void* buf = GetDataBuffer(size);
+    memcpy(buf, buffer, size);
+    ReportDataReceived(size);
+    return size;
+}
+
+size_t wxWebResponseCURL::AddHeaderData(const char * buffer, size_t size)
+{
+    wxString hdr(buffer, size);
+    hdr.Trim();
+
+    if ( hdr.StartsWith("HTTP/") )
+    {
+        // First line of the headers contains status text after
+        // version and status
+        m_statusText = hdr.AfterFirst(' ').AfterFirst(' ');
+        m_headers.clear();
+    }
+    else if ( !hdr.empty() )
+    {
+        wxArrayString hdrArr = wxSplit(hdr, ':');
```

Comment:
We should use simpler and more correct (in presence of more colons)

```
wxString hdrValue;
wxString hdrName = hdr.BeforeFirst(':', &hdrValue);
```
instead.